ext/liquid_c/tokenizer.c

```diff
@@ -66,20 +67,28 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
 
     token->str = cursor;
     token->type = TOKEN_RAW;
+    token->lstrip = 0;
+    token->rstrip = 0;
 
     while (cursor < last) {
         if (*cursor++ != '{')
             continue;
 
         char c = *cursor++;
+        char w = *cursor++;
         if (c != '%' && c != '{')
             continue;
-        if (cursor - tokenizer->cursor > 2) {
+        if (cursor - tokenizer->cursor > 3) {
             token->type = TOKEN_RAW;
-            cursor -= 2;
+            cursor -= 3;
+            token->rstrip = (w == '-');
```

Comment:
I would keep the lstrip and rstrip initialization, then replace the other above changes to this function with

```c
if (cursor <= last && *cursor == '-') {
  cursor++;
  token->rstrip = 1;
}
```